### ISSUE TITLE
`Lint/MixedCaseRange` error for invalid range

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#12843]
As shown below, a regex wouldn't normally be written this way, but it can occur when using an LSP.

## Expected behavior

Shouldn't raise.

## Actual behavior

```
undefined method `begin' for nil
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:105:in `rewrite_regexp_range'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:51:in `block (2 levels) in on_regexp'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/base.rb:392:in `correct'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/base.rb:181:in `add_offense'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:50:in `block in on_regexp'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:65:in `block (2 levels) in each_unsafe_regexp_range'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:60:in `reject'
/Users/andyw8/.gem/ruby/3.3.0/gems/rubocop-1.62.1/lib/rubocop/cop/lint/mixed_case_range.rb:60:in `block in each_unsafe_regexp_range'
/Users/andyw8/.gem/ruby/3.3.0/gems/regexp_parser-2.9.0/lib/regexp_parser/expression/methods/traverse.rb:75:in `block in each_expression_without_index'
```

## Steps to reproduce the problem

Run against this:
```
/[A-_]$/
```

## RuboCop version

```
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.3.0) [arm64-darwin23]
  - rubocop-minitest 0.35.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.8.0
```
